woocommerce/woocommerce
# Marketplace: update no results page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
We show two separate no results components on the search results page, this PR reduces it to one. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch and build changes
2. Go to any In App Marketplace page (Discover, Browse, Themes... doesn't matter). 
3. Perform a search where you'll get hits for both extensions and themes. 
    - You should see both results for extensions and themes on the search reuslts page. 
    - Clicking **View all** should show you the full result set. 
4. Perform a search where you'll get hits only for extensions.
    -  You'll only see the extensions component. There will no mention of themes.
    - The list will be expanded: there won't be any **View all** button and you'll see the category selector.
5. Perform a search where you'll get hits only for themes.
    -  You'll only see the themes component. There will no mention of extensions.
    - The list will be expanded: there won't be any **View all** button and you'll see the category selector.
6. Perform a search where you'll get **no** hits for extensions and themes.
    -   You'll see the new NoResults component.
    - You should see recommendations for both themes and extensions.
7. Browse **Browse** and **Themes** pages. Confirm you can use filters still. 
8. Please test around these changes.

### Screenshots
![image](https://github.com/woocommerce/woocommerce/assets/10389957/3d8292b5-ff77-4b5e-909a-f974afd52948)
